app/models/concerns/roleable.rb

```diff
@@ -18,6 +18,14 @@ def regular?
     !staff?
   end
 
+  def whisperer?
+    return @whisperer if defined?(@whisperer)
+    return @whisperer = false if !SiteSetting.enable_whispers?
+    return @whisperer = true if staff?
+    return @whisperer = false if SiteSetting.whispers_allowed_groups.blank?
+    @whisperer = group_users&.exists?(group_id: SiteSetting.Whispers.allowed_group_ids)
+  end
```

Comment:
```suggestion
  def whisperer?
    @whisperer ||= begin
      return false if !SiteSetting.enable_whispers?
      return true if staff?
      return false if SiteSetting.whispers_allowed_groups.blank?
      group_users&.exists?(group_id: SiteSetting.Whispers.allowed_group_ids)
    end
  end
```

Reply:
Note that you need to clear the `@whisperer` instance variable on `reload!`

Reply:
Also I realised there is a `User#primary_group_id` column. I think it'll be worth it to check that column against the group ids site settings first before querying the database.